src/services/generateFlightsByDate.ts

```diff
@@ -11,7 +11,7 @@ export function generateFlightsByDate(date: DateTime): Flight[] {
 
   // Test cache for data
   const cachedFlights = flightCache.getFlights(seed);
-  if (!cachedFlights) {
+  if (!cachedFlights|| cachedFlights.length === 0) {
```

Comment:
```suggestion
  if (!cachedFlights) {
```

Reply:
Without this check it would allow empty lists to count as a valid response from the cache and wouldn't generate any new flights